ros_cross_compile/ros_cross_compile.py

```diff
@@ -136,13 +153,31 @@ def cross_compile_pipeline(
     args: argparse.Namespace,
 ):
     platform = Platform(args.arch, args.os, args.rosdistro, args.sysroot_base_image)
+    dc = data_collector()
 
+    # define paths for reading/writing
     ros_workspace_dir = Path(args.ros_workspace)
+    metrics_dir = Path(str(ros_workspace_dir) + '/cc_internals/metrics_data/')
     skip_rosdep_keys = args.skip_rosdep_keys
     custom_data_dir = _path_if(args.custom_data_dir)
     custom_rosdep_script = _path_if(args.custom_rosdep_script)
     custom_setup_script = _path_if(args.custom_setup_script)
 
+    # set up class wrappers for functions
+    gather_rosdeps_wrapper = pipeline_stage("gather_rosdeps", gather_rosdeps)
+    rosdep_script_wrapper = pipeline_stage("assert_install_rosdep_script_exists",
+                                           assert_install_rosdep_script_exists)
```

Comment:
This really isn't a stage of the pipeline, it's a precondition for the "create_sysroot" step - as per our discussion we should bundle this into the step to check its prerequisites before running, and raise an error if it's not met.